web/api/servers.go

```diff
@@ -978,6 +989,215 @@ func editServerDataAdmin(c *gin.Context) {
+		return
+	}
+
+	c.Status(http.StatusOK)
+}
+
+// @Summary Restore backup
+// @Description Removes all exisiting files and restores the server to the state of the backup
+// @Success 204 {object} nil
+// @Param id path string true "Server ID"
+// @Param backupId path string true "Backup ID"
+// @Router /api/servers/{id}/backup/Delete/{backupId} [delete]
+// @Security OAuth2Application[server.backup.restore]
+func restoreBackup(c *gin.Context) {
+	server := getServerFromGin(c)
+	db := middleware.GetDatabase(c)
+	ns := &services.Node{DB: db}
+	bs := &services.Backup{DB: db}
+	node := &server.Node
+
+	var err error
+	var backupId uint
+	if backupId, err = cast.ToUintE(c.Param("backupId")); err != nil {
+		response.HandleError(c, err, http.StatusBadRequest)
+		return
+	}
+
+	backup, err := bs.GetById(backupId)
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+		return
+	}
+	if backup == nil {
+		c.Status(http.StatusNotFound)
+		return
+	}
+
+	resolvedPath := "/daemon/server/" + server.Identifier + "/backup/restore" + "?fileName=" + backup.FileName
+
+	callResponse, err := ns.CallNode(node, "POST", resolvedPath, nil, nil)
+	if callResponse.StatusCode == http.StatusBadRequest { //If its a local node, the err will not be set, have to check the status code
+		newHeaders := cleanHttpReturnErrors(callResponse.Header)
+
+		c.DataFromReader(callResponse.StatusCode, callResponse.ContentLength, callResponse.Header.Get("Content-Type"), callResponse.Body, newHeaders)
+		c.Abort()
+		return
+	}
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+		return
+	}
+
+	c.Status(http.StatusOK)
+}
+
+// @Summary Download backup
+// @Description Download a server backup
+// @Success 204 {object} nil
+// @Param id path string true "Server ID"
+// @Param backupId path string true "Backup ID"
+// @Router /api/servers/{id}/backup/Delete/{backupId} [delete]
+// @Security OAuth2Application[server.backup.restore]
+func downloadBackup(c *gin.Context) {
+	server := getServerFromGin(c)
+	db := middleware.GetDatabase(c)
+	ns := &services.Node{DB: db}
+	bs := &services.Backup{DB: db}
+	node := &server.Node
+
+	var err error
+	var backupId uint
+	if backupId, err = cast.ToUintE(c.Param("backupId")); err != nil {
+		response.HandleError(c, err, http.StatusBadRequest)
+		return
+	}
+
+	backup, err := bs.GetById(backupId)
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+		return
+	}
+	if backup == nil {
+		c.Status(http.StatusNotFound)
+		return
+	}
+
+	resolvedPath := "/daemon/server/" + server.Identifier + "/backup/download" + "?fileName=" + backup.FileName
+
+	callResponse, err := ns.CallNode(node, "GET", resolvedPath, nil, nil)
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+		return
+	}
+
+	newHeaders := cleanHttpReturnErrors(callResponse.Header)
+
+	c.DataFromReader(callResponse.StatusCode, callResponse.ContentLength, callResponse.Header.Get("Content-Type"), callResponse.Body, newHeaders)
+	c.Abort()
```

Comment:
Should not need an abort, this should be the last op

Reply:
fixed in [3afa64c](https://github.com/pufferpanel/pufferpanel/pull/1379/commits/3afa64cabbf2a66b294418fec744d28d5024bb52)